File: Recursion/198_HouseRobber.java
class Solution {
    public int rob(int[] nums) {
        return helper(nums,0);
    }
    public int helper(int []nums,int ind){
        if(ind>=nums.length){
            return 0;
        }
        int rc=helper(nums,ind+2)+nums[ind];
        int sc=helper(nums,ind+1);
        
        return Math.max(rc,sc);
    }
}

File: Recursion/198_HouseRobber.cpp
#include<iostream>
#include<vector>
#include<algorithm>
using namespace  std;
int helper(vector<int>nums,int index){
    if(index>=nums.size()){
        return 0;
    }
    int rc=helper(nums,index+2)+nums[index];
    int sc=helper(nums,index+1);

    return max(rc,sc);
}
int main(){
    vector<int>nums={5,6,9,7,1,34};
    cout<< helper(nums,0)<<endl;
}
